src/ioctl.rs

```diff
@@ -154,17 +198,100 @@ type IoctlRequest = c_int;
 #[cfg(not(target_env = "musl"))]
 type IoctlRequest = c_ulong;
 
-/// Run an ioctl with no arguments.
+/// Run an [`ioctl`](http://man7.org/linux/man-pages/man2/ioctl.2.html)
+/// with no arguments.
+///
+/// # Arguments
+///
+/// * `fd`: an open file descriptor corresponding to the device on which
+/// to call the ioctl.
+/// * `req`: a device-dependent request code.
+///
+/// # Safety
+///
+/// The caller should ensure to pass a valid file descriptor and have the
+/// return value checked.
+///
+/// # Examples
+///
+/// ```
+/// extern crate vmm_sys_util;
+/// #
+/// # use std::fs::File;
+/// # use std::os::raw::c_ulong;
+/// # use std::os::unix::io::FromRawFd;
+/// use vmm_sys_util::ioctl::ioctl;
+///
+/// const TUNSETOFFLOAD: c_ulong = 0x4004_54d0;
+/// let file = unsafe { File::from_raw_fd(0) };
```

Comment:
I just noticed that this example does not actually work. In the documentation, in the safety section you say that the caller should check the return value and that the fd should be valid. 

All the examples on ioctl functions will not work becasue:
- you are using the same ioctl, but that ioctl cannot be called with the parameters you are using
- you are not opening `/dev/net/tun` file

I suggest using kvm ioctls instead as it is easier to get the code working.

Reply:
One question is I found kvm-ioctls crate already has the same file as here. Do we still need re-publish it in this crate? I am thinking maybe other hypervisors would use...?

Reply:
Once we publish vmm-sys-utils to crates.io I will remove the `ioctl` file from kvm-ioctls and use the ioctl defines from here.

Reply:
Got it. Thanks for explanation.

Reply:
kvm-ioctl are used for examples. Updated.

Reply:
I didn't really mean to use the `kvm-ioctls` crate. That would result in a cycle dependency as `kvm-ioctls` will eventually use `vmm-sys-utils` for the ioctl macros and functions. I am sorry if I was not explicit enough. I meant using the ioctls on the KVM device. We can use the simplest ioctls we can find that don't need many defines.

I can add a commit on top of your commit because I feel this is wasting too many cycles. What do you say?

Reply:
Of course, I'm fine with that. 
But I think the tests for `ioctl()` and `ioctl_with_val()` are just simple tests for raw ioctl, there's no usage for other vmm-sys-utils in cycle.
And for others, since we want to pass the real and meaningful test parameters, those use `ioctl()` and `ioctl_with_val` in a cycle. Do you feel it does not make sense?

Reply:
I think is better to have no examples at all rather than examples that are not working. If you think adding examples that work is too complicated/adds too much code, we can make an exception for this module. What do you say?

Reply:
I feel fine for both adding or not. We can keep the ioctl and ioctl_with_val ones because they can give an example for ioctl series functions and are also meaningful,, if you feel other are too complicated.

Reply:
I personally don't think it is complicated. But you need to do a bit of research to understand which ioctls are appropriate for examples. If you don't know already some ioctls that would be a good match for the functions, I don't think it is worth it to invest time in it.

We can keep `ioctl` and `ioctl_with_val` only if you remove the dependencies on `kvm-bindings` and `kvm-ioctls`. This can be done by:
- use File::open("/dev/kvm") instead of Kvm::new()
- define KVM_CAP_USER_MEMORY in the example, don't use it from kvm-bindings.
- define KVM_GET_API_VERSION in the example, don't use it from kvm-bindings.